.github/workflows/release.yml

```diff
@@ -37,7 +37,7 @@ jobs:
           python -m build --sdist
           twine check --strict dist/*
 
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
```

Comment:
See the commit that changes this on main: https://github.com/shapely/shapely/commit/bc4632646009429952f344b8c7e11107c0fd513d

I think you will have to copy over the other changes as well to make this fully working (I assume github actions will also complain about the v3 in `actions/download-artifact@v3`)